lib/charms/alertmanager_k8s/v0/alertmanager_remote_configurer.py

```diff
@@ -0,0 +1,1940 @@
+
+    Args:
+        path: Alertmanager templates file path
+
+    Returns:
+        str: Alertmanager templates
+    """
+    path = Path(path)
+    if path.is_file():
+        with open(path, "r") as template_file:
+            templates = template_file.read()
+        return templates
+    else:
+        logger.warning(f"Given Alertmanager templates file {path} doesn't exist. Skipping...")
+        raise FileNotFoundError
+
+
+class AlertmanagerRemoteConfigurerProvider(Object):
+    def __init__(
+        self,
+        charm: CharmBase,
+        relation_name: str = DEFAULT_RELATION_NAME,
+        config_file_path: str = DEFAULT_ALERTMANAGER_CONFIG_FILE_PATH,
+        api_address: str = "http://localhost:9093",
+    ):
+        super().__init__(charm, relation_name)
+        self._charm = charm
+        self._relation_name = relation_name
+        self._config_file_path = config_file_path
+        self.api_address = api_address
+
+        on_relation = self._charm.on[self._relation_name]
+
+        self.framework.observe(on_relation.relation_joined, self._on_relation_joined)
+
+    def _on_relation_joined(self, event: RelationJoinedEvent) -> None:
+        """Callback function handling actions taken when `alertmanager-remote-config-write`
+        relation joins.
+
+        Args:
+            event: Juju RelationJoinedEvent
+
+        Returns:
+            None
+        """
+        self.load_initial_alertmanager_config_to_relation_data_bag(event)
+
+    def load_initial_alertmanager_config_to_relation_data_bag(
+        self, event: RelationJoinedEvent
+    ) -> None:
+        """Reads current Alertmanager configuration from alertmanager.yml and puts the content
+        inside relation data bag. If needed, it can be used by `alertmanager-remote-config-write`
+        consumer as a starting point for generating custom configuration.
+
+        Args:
+            event: Juju RelationJoinedEvent
+
+        Returns:
+            None
+        """
+        if not self._charm.unit.is_leader():
+            return
+        alertmanager_config = self._get_current_config()
+        event.relation.data[self._charm.app]["alertmanager_config"] = json.dumps(
+            alertmanager_config
+        )
+
+    def config(self) -> dict:
+        config = {}
+        remote_config_write_relation = self._charm.model.get_relation(self._relation_name)
+        if remote_config_write_relation:
+            try:
+                raw_config = remote_config_write_relation.data[remote_config_write_relation.app][  # type: ignore[index]  # noqa: E501
+                    "alertmanager_config"
+                ]
+                if self._config_is_valid(json.loads(raw_config)):
+                    config = raw_config
+            except KeyError:
+                logger.warning(
+                    "Remote config provider relation exists, but no config has been provided."
+                )
+        return config
+
+    def templates(self) -> list:
```

Comment:
It was done in the `alertmanager-k8s`. I've copied the current version of the lib.
Please keep in mind, that this lib hasn't even been properly imported yet, because it's not officially available yet.